h2o-algos/src/main/java/hex/modelselection/ModelSelectionUtils.java

```diff
@@ -1102,19 +1102,21 @@ public static PredNameMinZVal findNumMinZVal(List<String> numPredNames, List<Dou
      * 2. To figure out which coefficient name corresponds to which column name, we use the catOffsets which will tell
      * us how many enum levels are used in the glm model coefficients.  If the catOffset for the first coefficient
      * says 3, that means that column will have three enum levels represented in the glm model coefficients.
+     * 
+     * For categorical predictors with multiple enum levels, we will look at the max z-score.  This will show the best
+     * performing enum levels.  We will remove the enum predictor if its best z-score is not good enough when compared
+     * to the z-score of other predictors.
```

Comment:
@wendycwong just to clarify for myself:
What does the z-score mean here?

My presumption without going deeper in the PR is that the z-score is basically a p-value that corresponds to hypothesis "Is the coefficient different from zero?". Thus lower p-value means that the variable is more informative for the model, i.e., is more likely to be non-zero.

If this is true, then min z-score represents the best.

Reply:
My apologies @tomasfryda , it takes me a while to figure out what z-score is and then I expect you guys to know what that is.

Okay, for each coefficient, you want to see if it is significant or not?  A significant coefficient will have small p-value, the threshold is usually set to 0.05.  You are assuming a t-distribution or normal distribution depending on the number of data samples you have.  The p-value is calculated as P(x < Z) where P is either the t or normal probability mass function.  You will get a small p-value is |Z| is large.  The z-score is defined to be |Z| in our case.  The Z value is calculated as coefficient value/standard deviation of the coefficient.  

Therefore in this case, if I am looking at a set of coefficients and try to figure out which one to eliminate next, I will look at the coefficient with big p-values (small z-score or small |Z|).  If the p-value is big, we will not reject the null hypothesis that the coefficient is zero.  Hence, we will discard the coefficient with big p-value with confidence.

Regarding enum predictor, there are many enum levels.  Each enum level will have its own coefficient.  To decide if the enum predictor is bad, originally, I use the worst performing level z-score.  This is not very good because if there is one enum level that has small p-value (big |Z|), the whole enum predictor will be thrown away.  

In this fix, for enum predictor, I will use its best z-score (biggest value) and compare it to all other predictors z-value.  In this case, if an enum predictor is thrown away, we can be sure that none of its enum level has small p-value (big |Z|).  I think this makes more sense.

Please let me know if this is clear.

Reply:
Yes, this makes sense. Thank you @wendycwong ! I just wanted to make sure that we don't have incorrectly named variables (since the current implementation seems to do what you want to do in this PR if the z-value is actually p-value).